src/ingest-pipeline/airflow/dags/generate_usage_report.py

```diff
@@ -156,40 +156,47 @@ def build_report(**kwargs):
             count = count + 1
             task_id = specific_usage['taskid'][ind]
             lines_with_tid = []
+            hubmap_id = None
+            data_type = None
+            entity_type = None
+            remove_record = False
             dataset_id = None
             for line in full_contents:
                 index = line.find(task_id)
                 if index != -1:
                     lines_with_tid.append(line)
-
+            # hubmap public does not record dataset id, so for these transfers we indicate these are unknown.
+            public_id = "af603d86-eab9-4eec-bb1d-9d26556741bb"
+            is_public = False
+            if specific_usage["source_endpoint_id"][ind] == public_id or specific_usage["destination_endpoint_id"][ind] == public_id:
+                is_public = True
+                data_type = "Public: Unknown"
+                hubmap_id = "Public: Unknown"
+                entity_type = "Public: Unknown"
             # for every line that contained the task id we are looking for, we now
             # look for the reference number
-            for line in lines_with_tid:
-                log_entry = line.replace("-", "")
-                ref_index = log_entry.find("ref=") + 4
-                reference_number = log_entry[ref_index: ref_index + 32]
-                pattern = 'c_path=.*(\\w{32})'
-                ref_output = None
-                # once we have the reference number, we once again iterate through
-                # the file and look to find where it appears we are looking for the
-                # dataset id, which we locate using the above regex pattern
-                for b in full_contents:
-                    ref_index = b.find(reference_number)
-                    if ref_index == -1:
-                        matches = re.search(pattern, b)
-                        if matches is not None:
-                            ref_output = matches.group(1)
-                            break
-                if ref_output is not None:
-                    dataset_id = ref_output
-                    break
+            if is_public is not None:
```

Comment:
This validation will always be True since 'is_public' was just declared a couple of lines above with default value of False.

Reply:
Good catch, you are correct. In an earlier version of this script, is_public was initialized as None rather than false. When stitching together my revisions into the existing version in ingest-pipeline, I mistakenly left this in. Thank you for finding that. This has been fixed and I've pushed the fix to `Derek-Furst/update-generate-usage-reports`.